internal/app/structure.go

```diff
@@ -0,0 +1,136 @@
+	NameGraphQL          = "graphql"
+	NameIncomingWebhooks = "incoming_webhooks"
+	NameRealmConfig      = "realm_config"
+	NameRules            = "rules"
+	NameSecrets          = "secrets"
+	NameServices         = "services"
+	NameSource           = "source"
+	NameStitch           = "stitch"
+	NameTriggers         = "triggers"
+	NameValues           = "values"
+)
+
+// set of app structure files and directories
+var (
+	FileConfig      = File{Name: NameConfig, Ext: extJSON}
+	FileRealmConfig = File{Name: NameRealmConfig, Ext: extJSON}
+	FileSource      = File{Name: NameSource, Ext: extJS}
+	FileStitch      = File{Name: NameStitch, Ext: extJSON}
+
+	FileSecrets = File{Name: NameSecrets, Ext: extJSON}
+
+	DirAuthProviders = Dir{Name: NameAuthProviders}
+
+	DirFunctions = Dir{Name: NameFunctions}
+
+	DirGraphQL                = Dir{Name: NameGraphQL}
+	FileGraphQLConfig         = File{Parent: DirGraphQL, Name: NameConfig, Ext: extJSON}
+	DirGraphQLCustomResolvers = Dir{Parent: DirGraphQL, Name: NameCustomResolvers}
+)
+
+// FileAuthProvider creates the auth provider config filepath
+func FileAuthProvider(name string) File {
+	return File{Parent: DirAuthProviders, Name: name, Ext: extJSON}
+}
+
+func readAppStructureV1(appDir Directory) (map[string]interface{}, error) {
+	pkg := map[string]interface{}{}
+	if err := unmarshalAppConfigInto(appDir, &pkg); err != nil {
+		return nil, err
+	}
+	if err := unmarshalSecretsInto(appDir, pkg); err != nil {
+		return nil, err
+	}
+	if err := unmarshalDirectoryInto(appDir.Path, NameValues, pkg); err != nil {
+		return nil, err
+	}
+	if err := unmarshalDirectoryInto(appDir.Path, NameAuthProviders, pkg); err != nil {
+		return nil, err
+	}
+	if err := unmarshalFunctionsInto(appDir.Path, NameFunctions, pkg); err != nil {
+		return nil, err
+	}
+	if err := unmarshalDirectoryInto(appDir.Path, NameTriggers, pkg); err != nil {
+		return nil, err
+	}
+	if err := unmarshalGraphQLInto(appDir, pkg); err != nil {
+		return nil, err
+	}
+	if err := unmarshalServicesInto(appDir, pkg); err != nil {
+		return nil, err
+	}
+	return pkg, nil
+}
+
+func readAppStructureV2(appDir Directory) (map[string]interface{}, error) {
+	pkg := map[string]interface{}{}
+	if err := unmarshalAppConfigInto(appDir, &pkg); err != nil {
+		return nil, err
+	}
+	if err := unmarshalSecretsInto(appDir, pkg); err != nil {
+		return nil, err
+	}
+	if err := unmarshalDirectoryInto(appDir.Path, NameValues, pkg); err != nil {
+		return nil, err
+	}
+	if err := unmarshalDirectoryInto(appDir.Path, NameAuthProviders, pkg); err != nil {
+		return nil, err
+	}
+	if err := unmarshalFunctionsInto(appDir.Path, NameFunctions, pkg); err != nil {
+		return nil, err
+	}
+	if err := unmarshalDirectoryInto(appDir.Path, NameTriggers, pkg); err != nil {
+		return nil, err
+	}
+	if err := unmarshalGraphQLInto(appDir, pkg); err != nil {
+		return nil, err
+	}
+	if err := unmarshalServicesInto(appDir, pkg); err != nil {
+		return nil, err
+	}
+	return pkg, nil
```

Comment:
[q] is it intentional that this code duplicates the readAppStructureV1 function above?(apologies if I totally missed something here)

Reply:
Yeah, so this code can definitely still be considered a WIP for now (I meant to add a TODO, so thanks for asking this).

For some context, the new config version exports some data in different spots now (e.g. `sync.dev_mode_enabled` was moved into a `sync/config.json`), but the import route expects a different structure when processing requests.

The code paths are split here for now to support how the old CLI did it ("V1") and this new path ("V2") where I'm not _quite_ sure where it's headed still.  We have [a ticket](https://jira.mongodb.org/browse/REALMC-7865) planned to work out a lot of this transformation logic on server-side, so once that settles we should be able to easily update this and/or merge the two paths all together

Reply:
Perfect, that works for me thanks!